src/js/controllers/tab-receive.js

```diff
@@ -127,26 +108,47 @@ angular.module('copayApp.controllers').controller('tabReceiveController', functi
 
     $timeout(function() {
       $scope.$apply();
-    }, 100);
-
+    });
   });
 
-  $scope.updateCurrentWallet = function() {
+  function findWallet(wallet) {
+    return lodash.find($scope.wallets, function(w) {
+      return w.id == wallet.id;
+    });
+  };
+
+  $scope.loadAddresses = function(wallet, index) {
+    walletService.getAddress(wallet, false, function(err, addr) {
+      $scope.walletAddrs[wallet.id] = addr;
+    });
+  };
+
+  $scope.setWallet = function(index) {
+    $scope.wallet = $scope.wallets[index];
+    $scope.walletIndex = index;
+    if ($scope.walletAddrs[$scope.wallet.id].addr) $scope.addr = $scope.walletAddrs[$scope.walletIndex].addr;
+    else $scope.setAddress(false);
+    $timeout(function() {
+      $scope.$apply();
+    });
+  };
+
+  function updateCurrentWallet() {
     walletService.getStatus($scope.wallet, {}, function(err, status) {
       if (err) {
         return popupService.showAlert(bwcError.msg(err, gettextCatalog.getString('Could not update wallet')));
       }
-      $timeout(function() {
-        $scope.wallet = profileService.getWallet($scope.wallet.id);
-        $scope.wallet.status = status;
-        $scope.setAddress();
-        $scope.$apply();
-      }, 200);
+      $scope.wallet = profileService.getWallet($scope.wallet.id);
+      $scope.wallet.status = status;
+      $scope.setAddress();
     });
   };
 
   $scope.$on("$ionicView.beforeEnter", function(event, data) {
     $scope.wallets = profileService.getWallets();
+    if (!$scope.wallets[0]) return;
+    if ($scope.wallet && findWallet($scope.wallet))
+      $scope.setWallet($scope.wallets.indexOf($scope.wallet));
```

Comment:
Since you already have `$scope.wallet` why not just use it as the argument to `setWallet()`?